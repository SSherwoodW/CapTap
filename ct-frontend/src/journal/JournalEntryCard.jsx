import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Dialog } from '@headlessui/react';
import SaveToJournalModal from './SaveToJournalModal';
import CapTapApi from '../../../api';

function JournalEntryCard({ entry, onDelete }) {
    const { username } = useParams();

    const [isOpen, setIsOpen] = useState(false);
    const [updatedEntry, setUpdatedEntry] = useState(entry);


    useEffect(() => {
        setUpdatedEntry(entry);
    }, [entry]);

    const handleDelete = async () => {
        try {
            await onDelete(entry.id, username);
            setIsOpen(false); 
        } catch (error) {
            console.error('Error deleting journal entry:', error);
        }
    };

     const handleClose = () => {
        setIsOpen(false);
    };

    const handleOpen = () => {
        setIsOpen(true);
    };

    const updateDescription = (newDescription) => {
        const updatedEntry = { ...entry, description: newDescription };
        setUpdatedEntry(updatedEntry);
    };


    
    return (
        <div className="bg-gray-200 p-4 rounded-md shadow-md">
            <h2 className="text-lg font-semibold mb-2">{updatedEntry.description}</h2>
            <p className="text-sm mb-2">Range: {entry.range}</p>
            <div className="flex bg-gray-400 p-2 text-center rounded-t-sm justify-between items-center overflow-x-auto">
                {entry.journalPlayersData.length > 0 ? (
                <ul className="list-disc ml-6">
                    {entry.journalPlayersData.map((playerData, index) => (
                    <li key={index}>
                        Player ID: {playerData.playerId},{' '}
                        Stat Category: {playerData.statCategory},{' '}
                        Over/Under: {playerData.overUnder},{' '}
                        Value: {playerData.value}
                    </li>
                    ))}
                </ul>
                ) : (
                <p>No player data available</p>
                )}
            </div>
            <div className=''>
                <div className="">
                    <SaveToJournalModal entry={entry} updateDescription={ updateDescription} />
                </div>
                <div className="">
                    <button className="flex rounded-sm p-4 bg-red border-2 border-gray-400 text-gray-900 text-lg shadow-md hover:bg-gray-400 hover:ring-2 hover:ring-gray-300 focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                        onClick={() => setIsOpen(true)}>
                        Delete
                    </button>
                    <Dialog open={isOpen} onClose={handleClose}>
                        <Dialog.Overlay className="fixed inset-0 bg-black opacity-30" />
                        <div className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 bg-white p-4 rounded-lg">
                        <h2 className="text-lg font-semibold mb-4">
                            Are you sure you want to delete?
                        </h2>
                        <div className="flex justify-end">
                            <button onClick={handleClose} className="mr-4">Cancel</button>
                            <button onClick={handleDelete} className="bg-blue-500 text-white px-4 py-2 rounded">Delete</button>
                        </div>
                        </div>
                    </Dialog>
                </div>
            </div>
        </div>
    );
}

export default JournalEntryCard;
