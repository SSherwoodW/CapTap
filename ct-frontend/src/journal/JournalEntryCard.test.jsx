import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import JournalEntryCard from './JournalEntryCard';

const mockEntry = {
  description: 'Mock Entry',
  range: 'Mock Range',
  journalPlayersData: [
    { playerId: 1, statCategory: 'Mock Category 1', overUnder: 'Mock Over/Under 1', value: 'Mock Value 1' },
    { playerId: 2, statCategory: 'Mock Category 2', overUnder: 'Mock Over/Under 2', value: 'Mock Value 2' }
  ]
};

describe('JournalEntryCard', () => {
  it('renders correctly with provided entry data', () => {
    render(<JournalEntryCard entry={mockEntry} />);

    expect(screen.getByText((content, element) => {
      return (
        element.tagName.toLowerCase() === 'li' &&
        content.includes('Stat Category: Mock Category 1')
      );
    })).toBeInTheDocument();

    expect(screen.getByText((content, element) => {
      return (
        element.tagName.toLowerCase() === 'li' &&
        content.includes('Stat Category: Mock Category 2')
      );
    })).toBeInTheDocument();
  });

  it('renders "No player data available" message when no player data is provided', () => {
    const entryWithoutPlayerData = { ...mockEntry, journalPlayersData: [] };
    render(<JournalEntryCard entry={entryWithoutPlayerData} />);

    expect(screen.getByText('No player data available')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<JournalEntryCard entry={mockEntry} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
