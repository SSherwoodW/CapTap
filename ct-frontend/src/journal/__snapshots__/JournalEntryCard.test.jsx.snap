// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JournalEntryCard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-200 p-4 rounded-md shadow-md"
  >
    <h2
      class="text-lg font-semibold mb-2"
    >
      Mock Entry
    </h2>
    <p
      class="text-sm mb-2"
    >
      Range: Mock Range
    </p>
    <div
      class="flex bg-gray-400 p-2 text-center rounded-t-sm justify-between items-center overflow-x-auto"
    >
      <ul
        class="list-disc ml-6"
      >
        <li>
          Player ID: 1, Stat Category: Mock Category 1, Over/Under: Mock Over/Under 1, Value: Mock Value 1
        </li>
        <li>
          Player ID: 2, Stat Category: Mock Category 2, Over/Under: Mock Over/Under 2, Value: Mock Value 2
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
